testing/trino-product-tests/src/main/java/io/trino/tests/product/TestImpersonation.java

```diff
@@ -55,17 +58,52 @@ public class TestImpersonation
     @Named("databases.presto.configured_hdfs_user")
     private String configuredHdfsUser;
 
+    @Inject
+    @Named("databases.hive.warehouse_directory_path")
+    private String warehouseLocation;
+
     @BeforeMethodWithContext
     public void setup()
     {
         aliceExecutor = connectToTrino("alice@presto");
+        bobExecutor = connectToTrino("bob@presto");
     }
 
     @AfterMethodWithContext
     public void cleanup()
     {
         // should not be closed, this would close a shared, global QueryExecutor
         aliceExecutor = null;
+        bobExecutor = null;
+    }
+
+    @Test(groups = {HDFS_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testExternalLocationTableCreationFailure()
+    {
+        String tableNameBob = "bob_external_table";
+        String commonExternalLocationPath = warehouseLocation + "/nested_dir";
+        bobExecutor.executeQuery(format("DROP TABLE IF EXISTS %s", tableNameBob));
+        bobExecutor.executeQuery(format("CREATE TABLE %s (a bigint) WITH (external_location = '%s')", tableNameBob, commonExternalLocationPath + "/bob_table"));
+
+        String tableNameAlice = "alice_external_table";
+        aliceExecutor.executeQuery(format("DROP TABLE IF EXISTS %s", tableNameAlice));
+        assertQueryFailure(() -> aliceExecutor.executeQuery(format("CREATE TABLE %s (a bigint) WITH (external_location = '%s')", tableNameAlice, commonExternalLocationPath + "/alice_table")))
+                .cause()
+                .hasMessageContaining("Failed to create directory");
+    }
+
+    @Test(groups = {HDFS_NO_IMPERSONATION, PROFILE_SPECIFIC_TESTS})
+    public void testExternalLocationTableCreationSuccess()
+    {
+        String tableNameBob = "bob_external_table";
+        String commonExternalLocationPath = warehouseLocation + "/nested_dir";
+        bobExecutor.executeQuery(format("DROP TABLE IF EXISTS %s", tableNameBob));
+        bobExecutor.executeQuery(format("CREATE TABLE %s (a bigint) WITH (external_location = '%s')", tableNameBob, commonExternalLocationPath + "/bob_table"));
+
+        String tableNameAlice = "alice_external_table";
+        aliceExecutor.executeQuery(format("DROP TABLE IF EXISTS %s", tableNameAlice));
+        aliceExecutor.executeQuery(format("CREATE TABLE %s (a bigint) WITH (external_location = '%s')", tableNameAlice, commonExternalLocationPath + "/alice_table"));
+        assertThat(aliceExecutor.executeQuery(format("SELECT * FROM %s", tableNameAlice))).hasRowsCount(0);
```

Comment:
Can we also check the table owner in case of both impersonation enabled and disabled setup ?

Reply:
check added